yrs/src/block.rs

```diff
@@ -750,8 +789,16 @@ impl Item {
             }
 
             self.integrate_content(txn, pivot, &mut *parent_ref);
-            txn.add_changed_type(&mut *parent_ref, self.parent_sub.as_ref());
-            let parent_deleted = self.is_deleted();
+            txn.add_changed_type(&*parent_ref, self.parent_sub.as_ref());
+            let parent_deleted = if let TypePtr::Id(ptr) = &self.parent {
+                if let Some(item) = txn.store.blocks.get_item(ptr) {
+                    item.is_deleted()
+                } else {
+                    true
+                }
+            } else {
+                false
```

Comment:
This is accurate representation of corresponding Yjs code - previous implementation used simplification.